tools/serverpod_cli/test/generator/dart/client_code_generator/inheritance_class_test.dart

```diff
@@ -0,0 +1,234 @@
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+import 'package:serverpod_cli/src/generator/dart/client_code_generator.dart';
+import 'package:serverpod_cli/src/test_util/builders/class_definition_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:serverpod_cli/src/test_util/compilation_unit_helpers.dart';
+
+const projectName = 'example_project';
+final config = GeneratorConfigBuilder().withName(projectName).build();
```

Comment:
Question: How can these tests pass when we are not enabling the "extends" experiment?

Same for the `server_code_generator`.

Reply:
Uppon investigating it I would say, because we're guarding in `model_parser.dart` and here we create the models with `ClassDefinitionBuilder` and pass them directly to 

```dart
 var codeMap = generator.generateSerializableModelsCode(
      models: models,
      config: config,
  );
```

it works as intended in `inheritance_test.dart` because there we get the models from calling the `StatefulAnalyzer` 

```dart
var models = StatefulAnalyzer(config, modelSources, onErrorsCollector(collector)).validateAll();
```

Is this okay or should we add a guard also in the generator step?

Reply:
I think it is enough to guard against this in the parser, it is the only way a user can use the feature anyway. The code generator is a downstream consumer and shouldn't need to care. This looks good to me.